uk.co.compucorp.civicrm.tasksassignments/api/v3/Document.php

```diff
@@ -34,8 +34,17 @@ function civicrm_api3_document_create($params) {
     return $errors;
   }
 
+  // Processing for custom data
+  $customFields = _civicrm_api3_document_getcustomfieldsflipped();
+  foreach ($customFields as $key => $value) {
+    if (isset($params[$key])) {
+      $params[$value] = $params[$key];
```

Comment:
$customFields is an array of the form:
```
array(
  'custom_field_name' => 'custom_field_column',
);
```
This way, for example, if $params['remind_me'] is set (which is one of the custom fields used for documents), it will add something like $params['custom_4'] = $params['remind_me']. Now that $params['custom_4'] is set, when _civicrm_api3_custom_format_params() runs it will identify the field as a custom field and create an entry in $values array, which will then be added to $params['custom'], which is what will be finally be used in Activity's BAO to update custom fields.

https://github.com/civicrm/civicrm-core/blob/master/CRM/Activity/BAO/Activity.php#L521